LibSource/ComplexFloatArray.h

```diff
@@ -75,6 +75,29 @@ struct ComplexFloat {
     im = magnitude*sinf(phase);
   }
 
+  float getMagnitudeSquared() {
+    return re * re + im * im;
+  }
+
+  ComplexFloat getComplexConjugate() {
+    return ComplexFloat {re, -im};
+  }
+
+  ComplexFloat getDotProduct(ComplexFloat other) {
+    return ComplexFloat {re * other.re - im  * other.im, re * other.im + im * other.re};
+  }
+
+  /**
+   * Copy assignment
+   */
+  ComplexFloat& operator=(const ComplexFloat& other) {
```

Comment:
is this different from compiler generated default?

Reply:
Right, this is probably not needed.

However, I think this struct needs a constructor that accepts a single parameter and assigns this to both members. This is because CircularBuffer does this: https://github.com/pingdynasty/OwlProgram/blob/feature/audio-processors/LibSource/CircularBuffer.h#L190 . So currently making a buffer of complex floats would raise a compilation error.

If that would be added, there's no reason not to add a constructor that assigns both members to different values. Looks like both constructors should be defined as constexpr to ensure that ComplexFloat is a literal type.